archive/imo/imo2021_q1.lean

```diff
@@ -151,69 +151,158 @@ begin
   { refine ⟨126, 163, 198, _, _, _, _,⟨17, _⟩, ⟨18, _⟩, 19, _⟩; linarith },
 end
 
--- We will need a lemma, stating that if a given number a ∈ [n, 2n]
--- does not belong to A ⊆ [n, 2n], then it belongs to [n, 2n] \ A.
-lemma finset.Ico.mem_compl_iff_nmem_subset {a n : ℕ} (A : finset ℕ)
-  (h : a ∈ finset.Ico n (2 * n + 1)) : a ∉ A ↔ a ∈ finset.Ico n (2 * n + 1) \ A :=
+-- Since it will be more convenient to work with sets later on, we will translate the above claim
+-- to state that there always exists a set B ⊆ [n, 2n] of cardinality at least 3, such that each
+-- pair of pairwise unequal elements of B sums to a perfect square.
+lemma exists_finset_of_3_leq_card_with_pairs_summing_to_squares (n : ℕ) (hn : 100 ≤ n) :
+  ∃ (B : finset ℕ), 2 * 1 + 1 ≤ B.card ∧ (∀ (a b ∈ B), a ≠ b →
+  ∃ (k : ℕ), a + b = k * k) ∧ ∀ (c ∈  B), n ≤ c ∧ (c : ℕ) ≤ 2 * n :=
```

Comment:
No need for type annotations here. `B : finset ℕ`, so `c ∈ B` lets Lean know that `c : ℕ`
```suggestion
  ∃ k, a + b = k * k) ∧ ∀ c ∈ B, n ≤ c ∧ c ≤ 2 * n :=
```